core/assembly/MetadataLoadContext/Program.cs

```diff
@@ -0,0 +1,121 @@
+
+namespace MetadataLoadContextSample
+{
+    class Program
+    {
+        static int Main(string[] args)
+        {            
+            if (args.Length < 1)
+            {
+                Console.WriteLine("Usage: dotnet MetadataLoadContextSample.dll <assembly path>");
+                return 0;
+            }
+
+            string inputFile = args[0];
+
+            try
+            {
+                //get the array of runtime assemblies
+                //this will allow us to at least inspect types depending only on BCL
+                string[] runtimeAssemblies = Directory.GetFiles(RuntimeEnvironment.GetRuntimeDirectory(), "*.dll");
+
+                //create the list of assembly paths consisting of runtime assemblies and the input file
+                var paths = new List<string>(runtimeAssemblies);
+                paths.Add(inputFile);
+
+                //create MetadataLoadContext that can resolve assemblies using the created list
+                var resolver = new PathAssemblyResolver(paths);
+                var mlc = new MetadataLoadContext(resolver);
+
+                using (mlc)
+                {
+                    //load assembly into MetadataLoadContext
+                    Assembly assembly = mlc.LoadFromAssemblyPath(inputFile);
+                    AssemblyName name = assembly.GetName();
+
+                    //print assembly attribute information
+                    Console.WriteLine(name.Name + " has following attributes: ");
+
+                    foreach (CustomAttributeData attr in assembly.GetCustomAttributesData())
+                    {
+                        try
+                        {
+                            Console.WriteLine(attr.AttributeType);
+                        }
+                        catch (FileNotFoundException ex)
+                        {
+                            //we are missing the required dependency assembly
+                            Console.WriteLine("Error: " + ex.Message);
+                        }
+                    }
+
+                    Console.WriteLine();
+
+                    //print assembly type information
+                    Console.WriteLine(name.Name + " contains following types: ");
+
+                    foreach (TypeInfo t in assembly.GetTypes())
+                    {
+                        try
+                        {
+                            Type baseType = t.BaseType;
+
+                            if (t.IsClass)
+                            {
+                                Console.Write("class ");
+                            }
+                            else if (t.IsValueType)
+                            {
+                                if (String.Equals(baseType?.FullName, "System.Enum", StringComparison.InvariantCulture))
+                                {
+                                    Console.Write("enum ");
+                                }
+                                else
```

Comment:
Is struct really the only other option? Wouldn't it better to explicitly match it for future proofing?

Reply:
It is the only other option currently (technically built-in value types like `int` aren't structures by ECMA spec, but they are still declared as structs in corelib metadata). But i'll add a check that the struct is inherited from ValueType directly to future-proof against possible other special value types.